File: 2025/12/03/structured-outputs-llms-reliable-json/python/examples/task_triage_example.py
"""Complete task triage API example."""
import os
import sys
from dotenv import load_dotenv

# Add parent directory to path
sys.path.insert(0, os.path.join(os.path.dirname(__file__), '..'))

from src.schema import TaskTriage, Category
from src.prompt_builder import build_structured_prompt
from src.llm_client import StructuredLLM
from src.structured_output import get_structured_output

load_dotenv()


def main():
    """Run task triage example."""
    # Initialize LLM client
    llm = StructuredLLM(
        model="gpt-4",
        temperature=0.3
    )
    
    # Example issue descriptions
    issues = [
        "User reported that they cannot log in after resetting their password. This is blocking them from accessing their account.",
        "Feature request: Add dark mode to the mobile app",
        "Question: How do I export my data?",
        "The payment processing is broken and customers can't complete purchases. This is urgent."
    ]
    
    print("Task Triage Example\n" + "=" * 50)
    
    for issue in issues:
        print(f"\nIssue: {issue}")
        print("-" * 50)
        
        # Build prompt
        prompt = build_structured_prompt(
            schema=TaskTriage,
            input_text=issue,
            task_description="Categorize and prioritize the following issue.",
            examples=[
                {
                    "category": "bug",
                    "priority": 3,
                    "needs_human": True,
                    "summary": "User cannot log in after password reset"
                }
            ]
        )
        
        try:
            # Get structured output
            result = get_structured_output(
                llm=llm,
                prompt=prompt,
                schema=TaskTriage,
                max_retries=3
            )
            
            print(f"Category: {result.category.value}")
            print(f"Priority: {result.priority}")
            print(f"Needs Human: {result.needs_human}")
            if result.summary:
                print(f"Summary: {result.summary}")
            
        except Exception as e:
            print(f"Error: {e}")


if __name__ == "__main__":
    main()

File: 2025/12/03/structured-outputs-llms-reliable-json/typescript/examples/task-triage-example.ts
import 'dotenv/config';
import { StructuredLLM } from '../src/llm-client';
import { TaskTriageSchema } from '../src/schema';
import { buildStructuredPrompt } from '../src/prompt-builder';
import { getStructuredOutput } from '../src/structured-output';

async function main() {
  // Initialize LLM client
  const llm = new StructuredLLM(
    undefined, // Use OPENAI_API_KEY env var
    'gpt-4',
    30000,
    0.3
  );

  // Example issue descriptions
  const issues = [
    "User reported that they cannot log in after resetting their password. This is blocking them from accessing their account.",
    "Feature request: Add dark mode to the mobile app",
    "Question: How do I export my data?",
    "The payment processing is broken and customers can't complete purchases. This is urgent."
  ];

  console.log('Task Triage Example\n' + '='.repeat(50));

  for (const issue of issues) {
    console.log(`\nIssue: ${issue}`);
    console.log('-'.repeat(50));

    // Build prompt
    const prompt = buildStructuredPrompt(
      TaskTriageSchema,
      issue,
      [
        {
          category: 'bug',
          priority: 3,
          needs_human: true,
          summary: 'User cannot log in after password reset'
        }
      ],
      'Categorize and prioritize the following issue.'
    );

    try {
      // Get structured output
      const result = await getStructuredOutput(
        llm,
        prompt,
        TaskTriageSchema,
        3
      );

      console.log(`Category: ${result.category}`);
      console.log(`Priority: ${result.priority}`);
      console.log(`Needs Human: ${result.needs_human}`);
      if (result.summary) {
        console.log(`Summary: ${result.summary}`);
      }
    } catch (e: any) {
      console.error(`Error: ${e.message}`);
    }
  }
}

main().catch(console.error);

